### ISSUE TITLE
Signature verify issue when message length >32 characters

### ISSUE THREAD

**Issue by phillipmoran** [getamis/alice#116]
**Is this a BUG REPORT or FEATURE REQUEST?**:
BUG REPORT

**What happened**:
Using the example provided in alice, signing a message with 32 characters or less works fine. However, verifying a signed message of >32 characters fails.

**What you expected to happen**:
Expected to be able to successfully verify signature, regardless of message length.

**How to reproduce it (as minimally and precisely as possible)**:
Running in Python 3.6.5
verify succeeds for msg1, but fails for msg2.
```
from ecpy.curves     import Curve,Point
from ecpy.keys       import ECPublicKey, ECPrivateKey
from ecpy.ecdsa      import ECDSA
import ecpy
from hashlib         import sha256

x = 82413025421550975780465452441471161579661032066499417714228771730090744271913
y = 77571870843403463763751119133059698966486723073744225522787423143129206189894

r1 = 32695600450337335906409614685668869399586796812718975177037483691275486581146
s1 = 82626493698088701273493631120474244794368213158721442704511347257757089426086

r2 = 16059247622785756262933476254168544184547141890459315662517063489502002819228
s2 = 67558261624811019589051140398882107624531221420268322720544033809973798414645

msg1 = b'ABCDEFGHIJKLMNOPQRSTUVWXYZABCDEF'
msg2 = b'ABCDEFGHIJKLMNOPQRSTUVWXYZABCDEFG'

cv   = Curve.get_curve('secp256k1')
pu_key = ECPublicKey(Point(x, y, cv))

sig1 = ecpy.ecdsa.encode_sig(r1, s1, 'DER')
sig2 = ecpy.ecdsa.encode_sig(r2, s2, 'DER')

signer = ECDSA(fmt='DER')

print(signer.verify(msg1,sig1,pu_key)) # returns True
print(signer.verify(msg2,sig2,pu_key)) # returns False
```

Also ran an example from ECPy to make sure this wasn't an issue with the Python library. Here is that code below you can run if you'd like to double check yourself.
```
pu_key = ECPublicKey(Point(0x65d5b8bf9ab1801c9f168d4815994ad35f1dcb6ae6c7a1a303966b677b813b00,

                           0xe6b865e529b8ecbf71cf966e900477d49ced5846d7662dd2dd11ccd55c0aff7f,
                           cv))
pv_key = ECPrivateKey(0xfb26a4e75eec75544c0f44e937dcf5ee6355c7176600b9688c667e5c283b43c5,
                      cv)
msg3 = b'ABCDEFGHIJKLMNOPQRSTUVWXYZABCDEFG'
sig    = signer.sign(msg3, pv_key)
r3, s3 = ecpy.formatters.decode_sig(sig, fmt='DER')
sig3 = ecpy.ecdsa.encode_sig(r3, s3, 'DER')
print(signer.verify(msg3, sig3, pu_key)) # returns True
```

Is there any processing I need to do to longer messages to prepare it for tss signing?

Thanks for taking a look.

**Comment by cychuang0924** [comment]
Hi Phillipmoran,
       Assume that we focus on the case: secp256k1.
       Any input of our library should be the computing result of some "hash function" ( E.g. sha256 or Sha3) such that the size of input <= 32 byte.  In our library, we do not hash the input because there exists many methods to achieve  "hash a message to int". 
An example:    https://golang.org/src/crypto/ecdsa/ecdsa.go?s=8829:8889#L168

Go to your question, you should check that how to hash message to int in your verification function. 
In this case, you can use sha256 to hash the message, then you can pass the verification.
ref: https://ec-python.readthedocs.io/en/latest/_modules/ecpy/ecdsa.html#ECDSA.verify

If you have any problem, let me know please.






In our library, we do not hash the input because there exists many methods about "how to hash a message to int".
      
       


     

If you focus on the secp256k1, then you should find a method to 
Any input of our library should be the computing result of some hash function ( E.g. sha256 or Sha3) such that
the size of input <= 32 byte. 





In conclusion, if you want to use our library, you should hash the message by yourself.

**Comment by phillipmoran** [comment]
That makes sense, and solved my problem.
Thanks very much for your help :)
